app/lib/widgets/settings/share_settings.dart

```diff
@@ -0,0 +1,200 @@
+                child: Column(
+                  mainAxisAlignment: MainAxisAlignment.center,
+                  crossAxisAlignment: CrossAxisAlignment.stretch,
+                  children: [
+                    Column(
+                      crossAxisAlignment: CrossAxisAlignment.start,
+                      children: [
+                        TextFormField(
+                          decoration: const InputDecoration(
+                            icon: Icon(Icons.email),
+                            hintText: 'VanMoof account email',
+                            labelText: 'Email',
+                          ),
+                          onSaved: (value) {
+                            _email = value ?? '';
+                          },
+                          validator: (String? value) {
+                            return (value != null &&
+                                    !EmailValidator.validate(value))
+                                ? 'Invalid email address'
+                                : null;
+                          },
+                        ),
+                        Padding(
+                          padding: const EdgeInsets.only(top: 16.0),
+                          child: SizedBox(
+                            width: double.infinity,
+                            child: CupertinoSlider(
+                              value: _duration,
+                              min: 1.0,
+                              max: 365.0,
+                              onChanged: (newDuration) =>
+                                  setState(() => _duration = newDuration),
+                            ),
+                          ),
+                        ),
+                        Padding(
+                          padding: const EdgeInsets.symmetric(vertical: 6),
+                          child: ElevatedButton(
+                            onPressed: () => submitForm(Navigator.of(context)),
+                            child: Text('Share ${_duration.round()} days'),
+                          ),
+                        ),
+                        error != null
+                            ? Text(
+                                error!,
+                                style: TextStyle(
+                                    color: Theme.of(context).errorColor),
+                              )
+                            : Container(),
+                        success != null
+                            ? Text(
+                                success!,
+                                style: const TextStyle(color: Colors.green),
+                              )
+                            : Container(),
+                      ],
+                    ),
+                  ],
+                ),
+              ),
+            ],
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class _ShareHolderList extends State<ShareSettings> {
+  get api => obtainApiClient()!;
+  dynamic _shareHolders;
```

Comment:
Can you move functions below the class variables?

```suggestion
  dynamic _shareHolders;

  get api => obtainApiClient()!;
```